mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/StrictImapSearchAnalyzer.java

```diff
@@ -18,44 +18,40 @@
  ****************************************************************/
 package org.apache.james.mailbox.lucene.search;
 
-import java.io.Reader;
+import java.io.IOException;
 
 import org.apache.lucene.analysis.Analyzer;
-import org.apache.lucene.analysis.TokenStream;
-import org.apache.lucene.analysis.cn.smart.SentenceTokenizer;
-import org.apache.lucene.analysis.ngram.NGramTokenFilter;
+import org.apache.lucene.analysis.core.UpperCaseFilterFactory;
+import org.apache.lucene.analysis.core.WhitespaceTokenizerFactory;
+import org.apache.lucene.analysis.custom.CustomAnalyzer;
+import org.apache.lucene.analysis.ngram.NGramFilterFactory;
 
 /**
-*
-* {@link Analyzer} which match substrings. This is needed because of RFC 3501.
-* 
-* From RFC:
-* 
-*      In all search keys that use strings, a message matches the key if
-*      the string is a substring of the field.  The matching is
-*      case-insensitive.
-*
-*/
-public final class StrictImapSearchAnalyzer extends Analyzer {
-
-    private final int minTokenLength;
-    private final int maxTokenLength;
-    
-    public StrictImapSearchAnalyzer() {
-        this(3, 40);
-    }
-    
-    public StrictImapSearchAnalyzer(int minTokenLength, int maxTokenLength) {
-        this.minTokenLength = minTokenLength;
-        this.maxTokenLength = maxTokenLength;
+ * {@link Analyzer} which match substrings. This is needed because of RFC 3501.
+ * <p>
+ * From RFC:
+ * <p>
+ * In all search keys that use strings, a message matches the key if
+ * the string is a substring of the field.  The matching is
+ * case-insensitive.
+ */
+public final class StrictImapSearchAnalyzer {
+    static Analyzer getAnalyzer() {
+        return getAnalyzer(3, 40);
     }
 
-   /**
-    * @see org.apache.lucene.analysis.Analyzer#tokenStream(java.lang.String, java.io.Reader)
-    */
-   @Override
-   public TokenStream tokenStream(String fieldName, Reader reader) {
-       return new NGramTokenFilter(new UpperCaseFilter(new SentenceTokenizer(reader)), minTokenLength, maxTokenLength);
-   }
-   
+    static Analyzer getAnalyzer(int minTokenLength, int maxTokenLength) {
+        try {
+            return CustomAnalyzer.builder()
+                    .withTokenizer(WhitespaceTokenizerFactory.NAME)
+                    .addTokenFilter(UpperCaseFilterFactory.NAME)
+                    .addTokenFilter(NGramFilterFactory.NAME,
+                            "minGramSize", String.valueOf(minTokenLength),
+                            "maxGramSize", String.valueOf(maxTokenLength),
```

Comment:
Performance of this is likely to be terrible. Maybe we should not make this the default?

Reply:
What do you mean with performance? Creating the analyzer? I would change the analyzer to be a singleton.